omod/src/main/webapp/resources/dhisconnector-runreports.js

```diff
@@ -51,57 +59,95 @@ function onMappingSelect() {
-                weekEndDate = new Date(date.getFullYear(), date.getMonth(), date.getDate() - date.getDay() + 7)
+function handleYearChange(){
+    const yearSelector = $('#yearSelector');
+    let selectedYear = yearSelector.val();
+    const currentYear = moment().year();
+    if (selectedYear >= currentYear) {
+        yearSelector.val(currentYear - 1);
+        selectedYear = currentYear - 1;
+    }
+}
+
+function intializeDailyPicker(){
+    jQuery('#dailyPeriodSelector').datepicker({
+        dateFormat: 'yymmdd'
+    });
+    jQuery('#dailyPicker').show();
+    jQuery("datepicker").addEventListener("change", function() {
+        selectedPeriod = this.value;
+    });
+}
 
-                jQuery('#periodSelector').val(displayWeek);
+function initializeWeeklyPicker(){
+    jQuery('#weeklyPeriodSelector').datepicker({
+        showOtherMonths: true,
+        selectOtherMonths: false,
+        showWeek: true,
+        firstDay: 1,
+        onSelect: function (dateText, inst) {
+            let date = jQuery(this).datepicker('getDate');
+            let week = jQuery.datepicker.iso8601Week(date);
+            if (week < 10) {
+                let displayWeek = date.getFullYear() + "W0" + week;
+            } else {
+                let displayWeek = date.getFullYear() + "W" + week;
             }
-        });
 
-        jQuery('#ui-datepicker-div > table > tbody > tr').live('mousemove', function () {
-            jQuery(this).find('td a').addClass('ui-state-hover');
-        });
-        jQuery('.ui-datepicker-calendar tr').live('mouseleave', function () {
-            jQuery(this).find('td a').removeClass('ui-state-hover');
-        });
-    } else if (selectedMapping.periodType === 'Monthly') {
-        // set up monthly
-        jQuery('#periodSelector').monthpicker({
-            pattern: 'yyyymm'
-        });
-    } else {
-        // This is unknown
-        var text = 'DHIS period: ' + selectedMapping.periodType + ', Please type the appropriate value';
-        jQuery('#un-recognized-period').html(text);
-        toggleCustomRangeCheckbox(true);
-    }
+            weekStartDate = new Date(date.getFullYear(), date.getMonth(), date.getDate() - date.getDay() + 1);
+            weekEndDate = new Date(date.getFullYear(), date.getMonth(), date.getDate() - date.getDay() + 7)
+
+            jQuery('#periodSelector').val(displayWeek);
+        }
+    });
+    jQuery('#weeklyPicker').show();
+    jQuery("datepicker").addEventListener("change", function() {
+        selectedPeriod = this.value;
+    });
+
+    jQuery('#ui-datepicker-div > table > tbody > tr').live('mousemove', function () {
+        jQuery(this).find('td a').addClass('ui-state-hover');
+    });
+    jQuery('.ui-datepicker-calendar tr').live('mouseleave', function () {
+        jQuery(this).find('td a').removeClass('ui-state-hover');
+    });
+}
+
+function initaialzieMonthlyPicker(){
```

Comment:
a small spelling mistake:
```suggestion
function initializeMonthlyPicker(){
```